WordPress/src/main/java/org/wordpress/android/ui/activitylog/list/filter/ActivityLogTypeFilterViewModel.kt

```diff
@@ -41,46 +45,51 @@ class ActivityLogTypeFilterViewModel @Inject constructor(
     fun start(
         remoteSiteId: RemoteId,
         parentViewModel: ActivityLogViewModel,
-        initialSelection: List<Int>
+        dateRange: DateRange?,
+        initialSelection: List<String>
     ) {
         if (isStarted) return
         isStarted = true
         this.remoteSiteId = remoteSiteId
         this.parentViewModel = parentViewModel
         this.initialSelection = initialSelection
-
+        this.dateRange = dateRange
         fetchAvailableActivityTypes()
```

Comment:
Idea (💡): What if this function is transformed to `fetchAvailableActivityTypes(remoteSiteId, dateRange)`, accepting both parameters in its signature instead of depending on the view model state. Wdyt?

PS: By having the suggestion, the function becomes more pure, don't you think?

Reply:
Thanks for the comment ;). Hmm, I was thinking about this (I even tried to implement it), but I'm still not sure.

If the method wasn't updating an internal state of the VM (or it was public), I'd completely agree. However, since the method is updating `_uiState` + it's updating the db, I feel like this improvement loses its main benefits. Moreover, imo the code becomes a bit more difficult to follow since there are more fields/parameters.

In general, I'm a fan of passing whatever the method needs (and nothing more) into a method. But in this case it feels a bit weird. Having said all that, please do follow up if you think the code would be better. Perhaps, I'm missing something or I'm too used to a certain approach even though it's not good.

Reply:
Thanks for the response Jirka! 🤗 

I might be able to help you understand why I comment it in the first place and how I ended up to asking for this alternative. I saw the new `dateRange` field being added, then I saw that the `fetchAvailableActivityTypes()` function was the only one using it. Then I asked myself, why not remove the field and pass it to `fetchAvailableActivityTypes()`, only to realise that this function is used by another entry point. I then realised whey this state is needed.

Then, I thought, what if we make this more concrete and straightforward. In general having stateful class is something to be avoided, thus by making the usage of this state more evident it feels that it helps visualise the statefulness instead of hiding it within the internal of a function.

All the above is in addition to having a function as pure as possible. But, as I said, this is food for thought, just a conversation to have until next time.

Reply:
>  In general having stateful class is something to be avoided, 

💯 

> thus by making the usage of this state more evident it feels that it helps visualise the statefulness instead of hiding it within the internal of a function.

Interesting, I personally find it more evident when it's used directly. But I guess I see your point, if you find usages of the field, it's used just at a single place. However, if I was passing it as an argument, it'd be used at two places.

I'll need to think about this. I can see pros and cons of both approaches and I'm not sure yet, which one is better. Having said that, thank you for starting this discussion ;)!

Reply:
Thank YOU for the discussion, I love having those! ❤️